index.html

```diff
@@ -3861,8 +3861,8 @@ <h3>Algorithm</h3>
                     <li id="alg-compact-12_8_9_6" class="changed">Otherwise, if <var>container</var> includes <code>@index</code>
                       and <var>index key</var> is not <code>@index</code>:
                       <ol>
-                        <li>Reinitialize <var>container key</var> by <a>IRI compacting</a>
-                          <var>index key</var>.</li>
+                        <li id="alg-compact-12_8_9_6_1">Reinitialize <var>container key</var> by <a>IRI compacting</a>
+                          <var>index key</var> after first <a>IRI expanding</a> it.</li>
```

Comment:
An alternative to consider, which is probably more correct, would be to do the expansion of the `@index` value when the context is processed, as the value may expand differently than it would when compacting. It would still be a localized change, but not to the Compaction algorithm.

Reply:
I did try this, but remembering the original compacted term is necessary for expansion. The most correct solution would be to save the expanded version of this in the term definition, and use that in the Compaction algorithm instead of the original version, but that would be more disruptive to the rest of the document. This solution will solve any issues other than those specifically constructed to defeat it.